geopandas/_vectorized.py

```diff
@@ -1018,21 +1018,42 @@ def relate(data, other):
 
 
 def unary_union(data):
+    warning_msg = (
+        "unary_union resulted in an empty geometry and returned None. "
```

Comment:
I find the "resulted in an empty geometry and returned None" a bit unclear, since for the user, they just call `unary_union` attribute, and what it results in or returns is sounds basically as the same for them?

Reply:
Having just `"unary_union returned None"` would be better?

Reply:
Yes, that's all that the user sees: it returns None, and will return empty geometrycollection in the future. You could maybe mention this happens in case of empty or all-None GeoSeries.